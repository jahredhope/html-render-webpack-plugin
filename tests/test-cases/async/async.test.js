const MemoryFS = require("memory-fs");
const webpack = require("webpack");
const path = require("path");

const config = require("./webpack.config");
const HtmlRenderPlugin = require("../../../src");
const getDirContentsSync = require("../../utils/getDirContentsSync");

describe("Render asyncronously", () => {
  const renderDirectory = path.join(process.cwd(), "dist", "render");

  it("should render a HTML once resolved", async done => {
    const compiler = webpack(config(new HtmlRenderPlugin({ renderDirectory })));

    const memoryFs = new MemoryFS();
    compiler.outputFileSystem = memoryFs;

    compiler.run(() => {
      const contents = getDirContentsSync(renderDirectory, { fs: memoryFs });
      expect(contents).toMatchSnapshot();
      done();
    });
  });
  it("should render a multiple files at once", async done => {
    jest.setTimeout(1000);
    const compiler = webpack(
      config(
        new HtmlRenderPlugin({
          renderConcurrency: "parallel",
          routes: new Array(20).fill(null).map((_, i) => `page${i}`),
          renderDirectory
        })
      )
    );

    const memoryFs = new MemoryFS();
    compiler.outputFileSystem = memoryFs;

    compiler.run(error => {
      expect(error).toBe(null);
      const contents = getDirContentsSync(renderDirectory, { fs: memoryFs });
      expect(contents).toMatchSnapshot();
      done();
    });
  });
});
