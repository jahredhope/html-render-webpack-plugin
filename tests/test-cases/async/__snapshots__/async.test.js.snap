// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render asyncronously should render a HTML once resolved 1`] = `
Object {
  "index.html": "<html>
<body>
  Rendered with:&nbsp;
  <code>[
    {
      \\"route\\": \\"\\"
    }
  ]</code>
</body>
</html>",
}
`;

exports[`Render asyncronously should render a multiple files at once 1`] = `
Object {
  "page0": Object {
    "index.html": "<html>
<body>
  Rendered with:&nbsp;
  <code>[
    {
      \\"route\\": \\"page0\\"
    }
  ]</code>
</body>
</html>",
  },
  "page1": Object {
    "index.html": "<html>
<body>
  Rendered with:&nbsp;
  <code>[
    {
      \\"route\\": \\"page1\\"
    }
  ]</code>
</body>
</html>",
  },
  "page10": Object {
    "index.html": "<html>
<body>
  Rendered with:&nbsp;
  <code>[
    {
      \\"route\\": \\"page10\\"
    }
  ]</code>
</body>
</html>",
  },
  "page11": Object {
    "index.html": "<html>
<body>
  Rendered with:&nbsp;
  <code>[
    {
      \\"route\\": \\"page11\\"
    }
  ]</code>
</body>
</html>",
  },
  "page12": Object {
    "index.html": "<html>
<body>
  Rendered with:&nbsp;
  <code>[
    {
      \\"route\\": \\"page12\\"
    }
  ]</code>
</body>
</html>",
  },
  "page13": Object {
    "index.html": "<html>
<body>
  Rendered with:&nbsp;
  <code>[
    {
      \\"route\\": \\"page13\\"
    }
  ]</code>
</body>
</html>",
  },
  "page14": Object {
    "index.html": "<html>
<body>
  Rendered with:&nbsp;
  <code>[
    {
      \\"route\\": \\"page14\\"
    }
  ]</code>
</body>
</html>",
  },
  "page15": Object {
    "index.html": "<html>
<body>
  Rendered with:&nbsp;
  <code>[
    {
      \\"route\\": \\"page15\\"
    }
  ]</code>
</body>
</html>",
  },
  "page16": Object {
    "index.html": "<html>
<body>
  Rendered with:&nbsp;
  <code>[
    {
      \\"route\\": \\"page16\\"
    }
  ]</code>
</body>
</html>",
  },
  "page17": Object {
    "index.html": "<html>
<body>
  Rendered with:&nbsp;
  <code>[
    {
      \\"route\\": \\"page17\\"
    }
  ]</code>
</body>
</html>",
  },
  "page18": Object {
    "index.html": "<html>
<body>
  Rendered with:&nbsp;
  <code>[
    {
      \\"route\\": \\"page18\\"
    }
  ]</code>
</body>
</html>",
  },
  "page19": Object {
    "index.html": "<html>
<body>
  Rendered with:&nbsp;
  <code>[
    {
      \\"route\\": \\"page19\\"
    }
  ]</code>
</body>
</html>",
  },
  "page2": Object {
    "index.html": "<html>
<body>
  Rendered with:&nbsp;
  <code>[
    {
      \\"route\\": \\"page2\\"
    }
  ]</code>
</body>
</html>",
  },
  "page3": Object {
    "index.html": "<html>
<body>
  Rendered with:&nbsp;
  <code>[
    {
      \\"route\\": \\"page3\\"
    }
  ]</code>
</body>
</html>",
  },
  "page4": Object {
    "index.html": "<html>
<body>
  Rendered with:&nbsp;
  <code>[
    {
      \\"route\\": \\"page4\\"
    }
  ]</code>
</body>
</html>",
  },
  "page5": Object {
    "index.html": "<html>
<body>
  Rendered with:&nbsp;
  <code>[
    {
      \\"route\\": \\"page5\\"
    }
  ]</code>
</body>
</html>",
  },
  "page6": Object {
    "index.html": "<html>
<body>
  Rendered with:&nbsp;
  <code>[
    {
      \\"route\\": \\"page6\\"
    }
  ]</code>
</body>
</html>",
  },
  "page7": Object {
    "index.html": "<html>
<body>
  Rendered with:&nbsp;
  <code>[
    {
      \\"route\\": \\"page7\\"
    }
  ]</code>
</body>
</html>",
  },
  "page8": Object {
    "index.html": "<html>
<body>
  Rendered with:&nbsp;
  <code>[
    {
      \\"route\\": \\"page8\\"
    }
  ]</code>
</body>
</html>",
  },
  "page9": Object {
    "index.html": "<html>
<body>
  Rendered with:&nbsp;
  <code>[
    {
      \\"route\\": \\"page9\\"
    }
  ]</code>
</body>
</html>",
  },
}
`;
